backend/src/main/java/com/github/EazyErik/repository/CustomDrinksRepository.java

```diff
@@ -0,0 +1,19 @@
+package com.github.EazyErik.repository;
+
+
+import com.github.EazyErik.datalayer.CustomDrink;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+
+import java.security.Principal;
+import java.util.List;
+
+@Repository
+public interface CustomDrinksRepository extends MongoRepository<CustomDrink,String> {
+
+    List<CustomDrink> findAllByUsername(String username);
+    CustomDrink findCustomDrinkByCustomIDFromDB(String id);
```

Comment:
1/3
since `customIDFromDB` is marked with @Id in your Document, this line should be obsolete, since findById is always present. Also, find by something should always return an Optional, since you cant be sure to find something (otherwise null would be returned => could lead to NullPointerException), which is what the given findById already does.